Meter.c

```diff
@@ -380,7 +380,24 @@ Meter* Meter_new(const Machine* host, unsigned int param, const MeterClass* type
    if (Meter_initFn(this)) {
       Meter_init(this);
    }
-   Meter_setMode(this, type->defaultMode);
+
+   MeterModeId defaultMode = type->defaultMode;
+   uint32_t supportedMode = type->supportedMode ? type->supportedMode : METERMODE_DEFAULT_SUPPORTED;
+   assert(supportedMode);
+
+   if (!(supportedMode & (1 << defaultMode))) {
+      // The specified mode is not supported,
+      // Look up the next supported mode instead
+      uint32_t modeMask = (1ul << defaultMode) - 1;
+
+      uint32_t nextMode = supportedMode & ~modeMask;
+      nextMode = nextMode ? nextMode : supportedMode & modeMask;
+      assert(nextMode);
+
+      defaultMode = (MeterModeId)countTrailingZeros((uint32_t)nextMode);
```

Comment:
1. The check of `supportedMode` should be in `Meter_setMode()`, not here.
2. I would like to adjust the behavior a bit. When `Meter_setMode()` is given a mode that's unsupported by the meter, don't switch to the next mode automatically but just fail instead.  `Meter_setMode()` can be called when setting a meter from a configuration file, and for that case, the unsupported mode should fall back to the default, not the next supported mode.
3. The only case where we need to find ans set to the next supported mode is in `MetersPanel_eventHandler()`.
4. I might take my time to update my [meter-next-mode](https://github.com/Explorer09/htop-1/tree/meter-next-mode) branch to show my version of the idea. But not now.

Reply:
1. Can you make a small patch for 1.?

2. Also okay. Patch?

3. Okay. Any changes apart from the ones mentioned above?

4. Okay. Looking forward.

Reply:
@BenBE https://github.com/Explorer09/htop-1/commit/08053a4c464aeaa3b53257f77ff96f489fa2d6c9